### PR TITLE
MTDSA-1411 Add VAT organisation Auth logic

### PR DESCRIPTION
**Pull request by narendravijayarao**
Please check if this change includes (or even requires) the following:

 - [ ] unit tests
 - [ ] functional tests
 - [ ] audit events
 - [ ] RAML documentation
 - [ ] logging

### CODE REVIEW

File: app/uk/gov/hmrc/vatapi/resources/VatReturnsResource.scala [hunk 1]
```diff
@@ -68,7 +66,8 @@ object VatReturnsResource extends BaseController {
   }
 
   def retrieveVatReturns(vrn: Vrn, periodKey: String): Action[AnyContent] =
-    Action.async { implicit request =>
+    APIAction(vrn).async { implicit request =>
+      logger.debug(s"[VatReturnsResource] [retrieveVatReturns] Retrieve VAT returns for the VRN : $vrn")
```
Comment (MarkAKelly) [inline app/uk/gov/hmrc/vatapi/resources/VatReturnsResource.scala]: This is picking up the logger from BaseResource.  We can make this the standard logger or keep the resource specific logger they had created previously.

File: app/uk/gov/hmrc/vatapi/services/AuthorisationService.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
```
Comment (MarkAKelly) [inline app/uk/gov/hmrc/vatapi/services/AuthorisationService.scala]: Can we make this Object/Trait so we can Unit test this?

File: app/uk/gov/hmrc/vatapi/resources/wrappers/Response.scala [hunk 3]
```diff
@@ -51,6 +51,8 @@ trait Response {
 
   def underlying: HttpResponse
 
+  def json: JsValue = underlying.json
```
Comment (MarkAKelly) [inline app/uk/gov/hmrc/vatapi/resources/wrappers/Response.scala]: Has this snook back in?  This got replaced by the `jsonorerror`.

File: app/uk/gov/hmrc/vatapi/services/AuthorisationService.scala (hunk 2 above)
Comment (MarkAKelly) [reply app/uk/gov/hmrc/vatapi/services/AuthorisationService.scala]: (Can we unit test this)

**Comment by scottylace** [comment]
General comment, can we add a few negative auth component test scenarios

File: conf/application.conf [hunk 4]
```diff
@@ -238,13 +244,23 @@ Dev {
     test-scenario-simulation {
```
Comment (scottylace) [inline conf/application.conf]: Can we name this client-agent-simulation so its purpose is clearly known.

**Comment by scottylace** [comment]
Merged

**Pull request closed without merging**